### PR TITLE
Support slicing axis in Keras Tensor with MXNet backend

### PR DESCRIPTION
**Pull request by sandeep-krishnamurthy**
### Summary
* Support slicing the axis with '-1' indexing in tensors. Without this change we cannot slice a Keras Tensor with MXNet backend something like var[:, -1, :]
* Remove custom loss function warning, as the related MXNet slice and slice_axis issue https://github.com/awslabs/keras-apache-mxnet/issues/120 is resolved.

### Related Issues
* This PR will fix - https://github.com/awslabs/keras-apache-mxnet/issues/113
* This PR will fix - https://github.com/awslabs/keras-apache-mxnet/issues/139

### PR Overview

- [y] This PR requires new unit tests [y/n] (make sure tests are included)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n]

@roywei @kalyc

### CODE REVIEW

File: tests/keras/backend/backend_test.py [hunk 1]
```diff
@@ -1852,6 +1852,16 @@ def test_variable_support_bool_dtype(self):
             with pytest.raises(TypeError):
                 K.variable('', dtype='unsupported')
 
+    @pytest.mark.skipif((K.backend() != 'mxnet'),
+                        reason='Testing only for MXNet backend')
+    def test_tensor_slicing(self):
```
Comment (kalyc) [inline tests/keras/backend/backend_test.py]: Has this also been tested on the example submitted offline?

File: keras/backend/mxnet_backend.py [hunk 2]
```diff
@@ -3931,18 +3931,50 @@ def __getitem__(self, in_slice):
...
             else:
                 raise AttributeError('MXNet Backend: KerasSymbol __getitem__ error.')
-        return KerasSymbol(mx.sym.slice(self.symbol, begin=tuple(begin),
-                                        end=tuple(end)), neighbors=[self])
+        sliced_res = mx.sym.slice(self.symbol, begin=tuple(begin),
+                                  end=tuple(end))
+        if slice_axis:
+            num_sliced_axis = 0
+            for dim in sliced_dim:
+                sliced_res = mx.sym.squeeze(sliced_res, axis=dim - num_sliced_axis)
```
Comment (kalyc) [inline keras/backend/mxnet_backend.py]: [pep style off]

File: tests/keras/backend/backend_test.py (hunk 1 above)
Comment (sandeep-krishnamurthy) [reply tests/keras/backend/backend_test.py]: yes.

**Comment by sandeep-krishnamurthy** [comment]
@kalyc - Thanks for the review. Changed PR to dev. It was a miss to create PR on master.

File: keras/backend/mxnet_backend.py [hunk 3]
```diff
@@ -3931,18 +3931,50 @@ def __getitem__(self, in_slice):
...
+        #     1. After step 1, result = (2,1,3)
+        #     2. After step 2, result = (2,3) which is same as Numpy and other backend behavior.
+        slice_axis = False
+        sliced_dim = [d for d in range(len(in_slice)) if in_slice[d] == -1]
         for i in in_slice:
             if isinstance(i, int):
-                begin.append(i)
-                end.append(i + 1)
+                # Want to slice the complete axis
+                if i == -1:
```
Comment (roywei) [inline keras/backend/mxnet_backend.py]: Testing some corner cases and found out both numpy and keras-tf can handle slicing with negative index until the length of that axis, we should handle all negative index cases.

```
np_data = np.random.randn(5, 5, 5)
var = K.variable(np_data, dtype=np_data.dtype)
>>> np.array_equal(K.eval(var[:, 1, :]), np_data[:,1,:])
True
>>> np.array_equal(K.eval(var[:, -1, :]), np_data[:,-1,:])
True
>>> np.array_equal(K.eval(var[:, -2, :]), np_data[:,-2,:])
True
>>> np.array_equal(K.eval(var[:, -3, :]), np_data[:,-3,:])
True
>>> np.array_equal(K.eval(var[:, -4, :]), np_data[:,-4,:])
True
>>> np.array_equal(K.eval(var[:, -5, :]), np_data[:,-5,:])
True
```

Comment (sandeep-krishnamurthy) [reply keras/backend/mxnet_backend.py]: Good point. Thanks.
1. This cannot be supported with at Keras-MXNet backend level, because we cannot infer shape and validate if it is a PlaceHolder and slicing is in Batch dimension. 
2. -2, -3 and such indexing is very rare, and it is highly likely -1 is used for slicing the axis. Hence, I would rather leave this as not supported (till mxnet slicing supports it)

Let me know your suggestions.

Comment (roywei) [reply keras/backend/mxnet_backend.py]: Correct me if I am wrong: with the current code, following is the output of -3 index, you just need to squeeze it. The output is almost correct. 
```
>>> K.eval(var[:, -3, :])
array([[[ 1.05196469, -0.36612039,  0.55118293, -1.24901296,
          0.1445943 ]],

       [[ 0.38754396, -1.38874179, -1.73223019,  2.03238142,
          0.98131609]],

       [[-0.85328576, -0.18601252, -0.43296869,  0.19665155,
          0.11326693]],

       [[-0.64659878,  1.53067163,  0.28361218,  0.23279734,
         -0.198544  ]],

       [[ 0.26599282,  2.07316213, -0.2963127 ,  1.27382705,
         -0.93112912]]])
>>> np_data[:,-3,:]
array([[ 1.05196469, -0.36612039,  0.55118293, -1.24901296,  0.1445943 ],
       [ 0.38754396, -1.38874179, -1.73223019,  2.03238142,  0.98131609],
       [-0.85328576, -0.18601252, -0.43296869,  0.19665155,  0.11326693],
       [-0.64659878,  1.53067163,  0.28361218,  0.23279734, -0.198544  ],
       [ 0.26599282,  2.07316213, -0.2963127 ,  1.27382705, -0.93112912]])
>>> K.eval(var[:, -3, :]).shape
(5, 1, 5)
>>> np_data[:,-3,:].shape
(5, 5)

```

Comment (sandeep-krishnamurthy) [reply keras/backend/mxnet_backend.py]: You see that because, it is not actually slicing the axis as expected but treating it as begin=(None, -3, None), end = (None, -2, None).
So in effect MXNet is not understanding it needs to slice axis, hence the dimension is 1.
